doc/source/how-to-use-clientapp.rst

```diff
@@ -0,0 +1,304 @@
+
+        def set_parameters(self, parameters):
+            params_dict = zip(net.state_dict().keys(), parameters)
+            state_dict = OrderedDict({k: torch.tensor(v) for k, v in params_dict})
+            net.load_state_dict(state_dict, strict=True)
+
+        def fit(self, parameters, config):
+            self.set_parameters(parameters)
+            train(net, trainloader, epochs=1)
+            return self.get_parameters(config={}), num_examples["trainset"], {}
+
+        def evaluate(self, parameters, config):
+            self.set_parameters(parameters)
+            loss, accuracy = test(net, testloader)
+            return float(loss), num_examples["testset"], {"accuracy": float(accuracy)}
+
+The above is a familiar structure when implementing a custom client module using :code:`NumpyClient` (this has less boilerplate code to implement than :code:`Client`). 
+
+To use :code:`CifarClient()` with the high-level :code:`ClientApp` API, we need to do several things:
+
+#. Wrap :code:`CifarClient` in a function, such as :code:`client_fn`. This allows multiple instances of :code:`CifarClient` to be created when :code:`client_fn` is called.
+#. Convert :code:`CifarClient` to a :code:`Client` implementation using the :code:`to_client()` method.
+
+.. code-block:: python
+
+    # file: client.py
+
+    def client_fn() -> Client:
+        return CifarClient().to_client()
+
+Then, we create the :code:`ClientApp` as follows:
+
+.. code-block:: python
+
+    # file: client.py
+
+    app = ClientApp(client_fn=client_fn)
+
+Finally, your :code:`ClientApp` is ready to be executed!
+
+To run :code:`ClientApp` from CLI, use the |flower_clientapp_link|_ command. Pass the :code:`<module>:<attribute>` to the command, where :code:`module` is the filename (:code:`client.py`) and :code:`attribute` is the instantiated :code:`ClientApp` in the :code:`module`:
+
+.. code-block::shell
+
+    $ flower-client-app client:app  --insecure
+
+.. admonition:: Note
+    :class: note
+
+    In this example, the :code:`--insecure` command line argument starts Flower without HTTPS and is only used for prototyping. To run with HTTPS, we instead use the argument :code:`--certificates` and pass the paths to the certificates. Please refer to `Flower CLI reference <ref-api-cli.html>`_ for implementation details.
+
+As you can see, we can easily reuse existing Flower clients with the :code:`ClientApp` by adding 3 lines of code! Let’s now walk through how to use the low-level API for greater implementation flexibility in our projects.
+
+Using low-level API
+-------------------
+
+With Flower 1.8, we provide a set of low-level APIs to allow more versatile ways to implement any functionality that we like in the :code:`ClientApp`. The three functions that can be registered in a :code:`ClientApp` are:
+
+* :code:`app.train()`
+* :code:`app.evaluate()`
+* :code:`app.query()`
+
+which, you guessed it, runs the training, evaluation, and query on client-side data. 
```

Comment:
I would suggest to add some explanation about these. specially the query which is not explained at all..

Reply:
@mohammadnaseri Do you mean additional explainer text next to these bullet points? In each of the sections for `app.train()` and `app.evaluate()`, there are explanations for the methods, which I think are sufficient.

I've prioritized `app.train()` and `app.evaluate()` based on a discussion I had with @danieljanes.

Reply:
I meant mainly for `query` as it is mentioned but not explained. maybe we can remove it until we have some explanation for it.